oem/ibm/libpldmresponder/file_io_type_pcie.cpp

```diff
@@ -1105,5 +1106,24 @@ int PCIeInfoHandler::newFileAvailableWithMetaData(uint64_t /*length*/,
     return PLDM_ERROR_UNSUPPORTED_PLDM_CMD;
 }
 
+void PCIeInfoHandler::deleteTopologyFiles()
+{
+    if (receivedFiles.empty())
+    {
+        try
+        {
+            for (auto& path : fs::directory_iterator(pciePath))
+            {
+                fs::remove_all(path);
+            }
+        }
```

Comment:
why to use an iterator if you are using remove_all ?
can't we just do remove_all(pciePath) ?

Reply:
and also remove_all() return 0 if path does not exist, do you wanna check that and return as well ? incase some one deletes that directory accidentally.

Reply:
remove_all(pciePath) will delete the directory with its content . To to avoid creation of directory multiple times I used iterator to just delete the directory content.

Reply:
ack